src/test/java/api/loans/scenarios/CheckInDeclaredLostItemTest.java

```diff
@@ -0,0 +1,477 @@
+
+    verifyLostItemFeeAccount(isAccountClosed(itemFee));
+    verifyLostItemFeeAccountAction(isCloseActionCreated(itemFee));
+
+    verifyLostItemProcessingFeeAccount(isAccountClosed(processingFee));
+    verifyLostItemProcessingFeeAccountAction(isCloseActionCreated(processingFee));
+  }
+
+  @Test
+  public void canRefundTransferredFee() {
+    final double setCostFee = 10.89;
+    final double processingFee = 5.00;
+
+    useChargeableRefundableLostItemFee(setCostFee, processingFee);
+
+    declareItemLost();
+
+    final JsonObject lostItemFeeAccount = feeFineAccountFixture
+      .getLostItemFeeAccount(loan.getId());
+    feeFineAccountFixture.transfer(lostItemFeeAccount.getString("id"), setCostFee);
+
+    checkInFixture.checkInByBarcode(item);
+
+    verifyLostItemFeeAccount(isAccountRefundedFully(setCostFee));
+    verifyLostItemFeeAccountAction(areTransferRefundActionsCreated(setCostFee));
+
+    verifyLostItemProcessingFeeAccount(isAccountClosed(processingFee));
+    verifyLostItemProcessingFeeAccountAction(isCloseActionCreated(processingFee));
+  }
+
+  @Test
+  public void canRefundPaidFee() {
+    final double setCostFee = 9.99;
+    final double processingFee = 5.00;
+
+    useChargeableRefundableLostItemFee(setCostFee, processingFee);
+
+    declareItemLost();
+
+    final JsonObject lostItemFeeAccount = feeFineAccountFixture
+      .getLostItemFeeAccount(loan.getId());
+    feeFineAccountFixture.pay(lostItemFeeAccount.getString("id"), setCostFee);
+
+    checkInFixture.checkInByBarcode(item);
+
+    verifyLostItemFeeAccount(isAccountRefundedFully(setCostFee));
+    verifyLostItemFeeAccountAction(arePaymentRefundActionsCreated(setCostFee));
+
+    verifyLostItemProcessingFeeAccount(isAccountClosed(processingFee));
+    verifyLostItemProcessingFeeAccountAction(isCloseActionCreated(processingFee));
+  }
+
+  @Test
+  public void canRefundPaidAndTransferredFee() {
+    final double transferAmount = 6.0;
+    final double paymentAmount = 4.0;
+    final double setCostFee = transferAmount + paymentAmount;
+    final double processingFee = 5.00;
+
+    useChargeableRefundableLostItemFee(setCostFee, processingFee);
+
+    declareItemLost();
+
+    final JsonObject lostItemFeeAccount = feeFineAccountFixture
+      .getLostItemFeeAccount(loan.getId());
+    feeFineAccountFixture.transfer(lostItemFeeAccount, transferAmount);
+    feeFineAccountFixture.pay(lostItemFeeAccount, paymentAmount);
+
+    checkInFixture.checkInByBarcode(item);
+
+    verifyLostItemFeeAccount(isAccountRefundedFully(setCostFee));
+    verifyLostItemFeeAccountAction(areTransferRefundActionsCreated(transferAmount));
+    verifyLostItemFeeAccountAction(arePaymentRefundActionsCreated(paymentAmount));
+
+    verifyLostItemProcessingFeeAccount(isAccountClosed(processingFee));
+    verifyLostItemProcessingFeeAccountAction(isCloseActionCreated(processingFee));
+  }
+
+  @Test
+  public void canCloseAndRefundPaidAndTransferredFee() {
```

Comment:
Does close here refer to the loan or the fee? If so, is this specific to a fee that has been partly paid and partly transferred?

```suggestion
  public void shouldCloseAndRefundPaidAndTransferredFee() {
```

Reply:
This test is about refunding partial transfer/payment and closing the remaining amount.

Reply:
What is mean by closing the remaining amount, is that the same as cancelling?

Reply:
yes, sorry. It should be `cancel` instead. Hmm... in a fact it should be something like `should Cancel Remaining Refund Paid And Transferred And Close Fee`

Reply:
@bohdan-suprun No need to apologise. I've made a similar suggestion elsewhere. I'll let you decide and then all of these overlapping comments can be resolved.